### ISSUE TITLE
mpmath ascribes a value to divergent Lerch Phi series.

### ISSUE THREAD

**Issue by kspalaiologos** [mpmath/mpmath#676]
To quote some code from the mpmath implementation:

```python
    if z == 0:
        return a ** (-s)
    # Faster, but these cases are useful for testing right now
    if z == 1:
        return ctx.zeta(s, a)
    if a == 1:
        return ctx.polylog(s, z) / z
```

it is apparent that `Re(s)>1` condition is not checked. As such, per Wolfram, `lerchphi(1,0.5,1)` is complex infinity, while mpmath outputs the following:

```
>>> lerchphi(1, 0.5, 1)
mpf('-1.4603545088095868')
```

**Comment by fredrik-johansson** [comment]
This corresponds to the usual analytic continuation of the Riemann zeta function. Is there a strong reason to disallow such input?

**Comment by skirpichev** [comment]
> This corresponds to the usual analytic continuation of the Riemann zeta function.

@fredrik-johansson, are you sure?  According to the DLMF 25.14.2 - it's not: ``zeta(s, a)`` could be used only for certain conditions.  Ditto for the polylog in the next if case.

**Comment by fredrik-johansson** [comment]
It says that the identity should hold when those conditions are met. It doesn't say that the same identity can't hold when those conditions aren't met.

It's just a matter of convention how you define analytic continuations of functions of several variables; the question is what convention is more useful here.

**Comment by kspalaiologos** [comment]
Could we at least have this behaviour explicitly mentioned in the documentation then, if it is indeed intended?

**Comment by skirpichev** [comment]
See also https://github.com/sympy/sympy/issues/26707

**Comment by skirpichev** [comment]
On another hand, in neighborhood of z=1 the lerchphi() looks as [HurwitzLerchPhi](https://reference.wolfram.com/language/ref/HurwitzLerchPhi.html):
```
In [5]: lerchphi(1,0.5,1)
Out[5]: mpf('-1.4603545088095868')

In [6]: lerchphi(1+1e-10,0.5,1)
Out[6]: mpc(real='-1.4603545086843399', imag='-177245.3777445842')

In [7]: lerchphi(1-1e-10,0.5,1)
Out[7]: mpf('177243.91741666207')
```
c.f.
```
$ wolframscript
Wolfram Language 14.1.0 Engine for Linux x86 (64-bit)
Copyright 1988-2024 Wolfram Research, Inc.

(...)

In[17]:= HurwitzLerchPhi[1+10.^(-10),0.5,1]

Out[17]= -1.46035 - 177245. I

In[18]:= HurwitzLerchPhi[1-10.^(-10),0.5,1]

                            -11
Out[18]= 177244. - 1.5412 10    I

In[19]:= HurwitzLerchPhi[1,0.5,1.1]

Out[19]= ComplexInfinity
```

Certainly, I've failed to find a good source to validate that current mpmath's behaviour is "the *usual* analytic continuation of the Riemann zeta function".  Mathemaica output makes more sense for me.

CC @oscarbenjamin

BTW, recently added to the python-flint lerch_phi() looks differently:
```
In [9]: flint.acb(1+1e-10,0).lerch_phi(0.5,1)
Out[9]: [+/- 2.08e+4] + [-2e+5 +/- 4.36e+4]j

In [10]: flint.acb(1-1e-10,0).lerch_phi(0.5,1)
Out[10]: [2e+5 +/- 4.36e+4]
```

**Comment by oscarbenjamin** [comment]
> BTW, recently added to the python-flint lerch_phi() looks differently:

It looks to me that mpmath and python-flint (arb) are consistent here:
```python
In [6]: flint.acb(1,0).lerch_phi(0.5,1)
Out[6]: [-1.46035450880959 +/- 3.20e-15]

In [7]: mpmath.lerchphi(1, 0.5, 1)
Out[7]: mpf('-1.4603545088095868')

In [8]: eps = 1e-10

In [9]: flint.acb(1-eps,0).lerch_phi(0.5,1)
Out[9]: [2e+5 +/- 4.36e+4]

In [10]: mpmath.lerchphi(1-eps, 0.5, 1)
Out[10]: mpf('177243.91741666207')

In [11]: flint.acb(1+eps,0).lerch_phi(0.5,1)
Out[11]: [+/- 2.08e+4] + [-2e+5 +/- 4.36e+4]j

In [13]: mpmath.lerchphi(1+eps, 0.5, 1)
Out[13]: mpc(real='-1.4603545086843399', imag='-177245.3777445842')
```
The mpmath values seem to be within the ranges indicated by arb. We can try with more precision:
```python
In [14]: flint.ctx.prec = 100

In [15]: flint.acb(1+eps,0).lerch_phi(0.5,1)
Out[15]: [-1.46035450868434 +/- 5.62e-15] + [-177245.3777445841932 +/- 9.64e-15]j

In [16]: mpmath.lerchphi(1+eps, 0.5, 1)
Out[16]: mpc(real='-1.4603545086843399', imag='-177245.3777445842')
```
These look close although maybe a small error:
```python
In [22]: a = flint.acb(1+eps,0).lerch_phi(0.5,1)

In [23]: b = mpmath.lerchphi(1+eps, 0.5, 1)

In [24]: a.overlaps(b)
Out[24]: False

In [25]: mpmath.mp.prec = 100

In [26]: b = mpmath.lerchphi(1+eps, 0.5, 1)

In [27]: a.overlaps(b)
Out[27]: True
```

**Comment by skirpichev** [comment]
> The mpmath values seem to be within the ranges indicated by arb.

Yes, but real component in the second example seems too imprecise.  But they are consistent, indeed.

BTW, what do you think about OP issue?  E.g. I think that at z=1 should be singularity, as Mathematica thinks.  Instead, we have (both in mpmath and sympy/diofant):
```
In [1]: lerchphi(1, 1/2, 1)
Out[1]: Φ(1, 0.5, 1)

In [2]: lerchphi(1, S(1)/2, 1)
Out[2]: Φ(1, 1/2, 1)

In [3]: _.evalf()
Out[3]: -1.46035450880959

In [4]: expand_func(_2)
Out[4]: ζ(1/2)

In [5]: _.evalf()
Out[5]: -1.46035450880959
```

**Comment by oscarbenjamin** [comment]
It all seems consistent as far as I can tell so is this just a question of which definition is in use?

> > The mpmath values seem to be within the ranges indicated by arb.
> 
> Yes, but real component in the second example seems too imprecise. But they are consistent, indeed.

An advantage of returning an interval is that you can return a less accurate result without compromising on rigour. The idea with `arb` is that making the interval small so that the relative error is `2**-prec` is only a soft goal. Instead the goals in decreasing order are:

1. The returned result always encloses the true answer (this is the only hard requirement).
2. The `prec` parameter gives some sort of bound on the computational effort so that increasing it takes longer but gives smaller intervals.
3. The width of the interval corresponds to the sort of accuracy that is implied by `prec` bits.

If you want to guarantee an accuracy of `b` bits then you need to call arb with increasing `prec` until the returned interval is as narrow as desired. This is shown in the python-flint docs where `showgood` is used to get a specified number of digits. Doing this sacrifices the bounds on computational effort though.

You have to choose between bounding computational effort or bounding the accuracy of the output because bounding either forces the other to be unbounded. The advantage of bounding computational effort is that it is easy to implement accuracy bounds on top (e.g. `showgood`) but the other way around does not work: if computational effort is not bounded then you don't have the opportunity to go back and ask for a less accurate result.

**Comment by skirpichev** [comment]
Documentation says: "The Lerch transcendent generalizes the Hurwitz zeta function :func:`zeta` (`z = 1`) and the polylogarithm :func:`polylog` (`a = 1`).".  So, analytic continuation seems to be documented for these cases.
